Sources/PrivacyStats/internal/CurrentPack.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  CurrentPack.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Combine
+import Foundation
+import os.log
+
+/**
+ * This actor provides thread-safe access to an instance of `PrivacyStatsPack`.
+ *
+ * It's used by `PrivacyStats` class to record blocked trackers that can possibly
+ * come from multiple open tabs (web views) at the same time.
+ */
+actor CurrentPack {
+    /**
+     * Current stats pack.
+     */
+    private(set) var pack: PrivacyStatsPack
+
+    /**
+     * Publisher that fires events whenever tracker stats are ready to be persisted to disk.
+     *
+     * This happens after recording new blocked tracker, when no new tracker has been recorded for 1s.
+     */
+    nonisolated private(set) lazy var commitChangesPublisher: AnyPublisher<PrivacyStatsPack, Never> = commitChangesSubject.eraseToAnyPublisher()
+
+    nonisolated private let commitChangesSubject = PassthroughSubject<PrivacyStatsPack, Never>()
+    private var commitTask: Task<Void, Never>?
+    private var commitDebounce: UInt64
+
+    /// The `commitDebounce` parameter should only be modified in unit tests.
+    init(pack: PrivacyStatsPack, commitDebounce: UInt64 = 1_000_000_000) {
+        self.pack = pack
+        self.commitDebounce = commitDebounce
+    }
+
+    deinit {
+        commitTask?.cancel()
+    }
+
+    /**
+     * This function is used when clearing app data, to clear any stats cached in memory.
+     *
+     * It sets a new empty pack with the current timestamp.
+     */
+    func resetPack() {
+        resetStats(andSet: Date.currentPrivacyStatsPackTimestamp)
```

Comment:
Great point - it actually should. It works because it's used by the Fire button which closes everything and opens a new window in which case new data is pulled, but overall this should trigger a notification. Thanks for spotting it!